### DISCUSSION TITLE
[enhancement][question] self-hosting server app on shared hosting provider

### DISCUSSION THREAD

**Discussion by TesterEU** [ente-io/ente#1816] [Enhancements]
In this discussion : [Allow for custom server for frontend apps](https://github.com/ente-io/ente/discussions/504) I received a reply from @mnvr :

> This thread is not the correct place to ask general self hosting questions, you end up notifying all the people who're watching it for updates on the original issue. If people keep doing this we'll have to lock this thread.
>
> About your question itself - If you're a non-technical person, I generally would not recommend trying to self host, it is not in a state where it is seamless enough (we're hoping to improve that, cf see this https://github.com/ente-io/ente/issues/983#issuecomment-2054808064). Immich, for example, might be a better option.
>
> About the shared hosting provider - From your description it doesn't seem like they give the ability to run Docker images, in that case unfortunately you won't be able to self host on that provider. It sounds like they setup they provide is for PHP apps.

1. When I said I don't have enough technical skills, I meant: Fork Your Repository and adapt it to the hosting environment.

2. I want to use your application because it is great. I appreciate the great work that has been done

3. What server applications require the Ente server app?
(My hosting provider gives me access to new php, apache, MySQL, Composera e.g laravel SSL, Direct Admin, etc)

4. You already have docker version of the server app (Ente). Please, Consider the self-hosted version (if some hosting meets the above requirements). I know my request for a self-hosted version of Your app is expensive, but look at it this way:
- You can grow the number of your application's users with a smaller budget. Hosting is much cheaper than VPS.
- Such applications already exist (but I liked your application better) - Piwigo:
Docker: https://hub.docker.com/r/linuxserver/piwigo
Self-Hosted: https://github.com/Piwigo/Piwigo?tab=readme-ov-file

I hope my request wont be taken as a demand. I wanted to know if this is possible? I respect your opinion and decision (whatever it may be) so let's discuss :)

P.S. Maybe this will add a new point: "self-hosted" to the table: [Ente_vs_GPotos](https://ente.io/compare/ente-vs-google-photos/)

**Comment by mnvr** [comment]
> I want to use your application because it is great. I appreciate the great work that has been done

Thank you. I was taking it that way only, my apologies if I came across otherwise. It is indeed a great feeling to know that I am working on something that is useful to people enough for them to spend the effort to try and self host.

> (My hosting provider gives me access to new php, apache, MySQL, Composera e.g laravel SSL, Direct Admin, etc). You already have docker version of the server app (Ente). Please, Consider the self-hosted version (if some hosting meets the above requirements). I know my request for a self-hosted version of Your app is expensive, but look at it this way:...

While I don't necessarily agree with all your points, but I agree with your overall conclusion - it would be so much nicer if more people could use/experience Ente. So no disagreements there. However, what you're requesting is currently just not feasible. The setup that your provider gives you limits it to traditional LAMP applications, and Ente has a very different architecture. Rewriting the entire server in php (it is currently in go) is not something that is going to happen at this point.

> I wanted to know if this is possible?

So it is not possible, unfortunately.

In any case, thank you for the suggestions and generally for checking Ente out. Hopefully the stars align better sometime soon in the future! ☺️

**Reply by TesterEU** [reply]
@mnvr Thank You for your response. In the future, it would be nice to hear that Ente (while maintaining the current version) has app for the PHP server (LAMP). Please keep this discussion and, if possible, add this as a milestone on project (e.g.new repository) roadmap (even if it is a long time project in future). I believe that you will be able to make Ente available to other interested people because this project deserves it 🥇

**Reply by pL-uTo** [reply]
@mnvr Do you have any plans to extend the application to the LAMP stack in the near future?

**Comment by MarkusDeland** [comment]
I'm also interested in Ente for the LAMP stack.

**Reply by 5murfete** [reply]
I'm too 🥵

**Reply by Raph-dK** [reply]
Piwigo alternative?

**Reply by ArataKuge** [reply]
there is interest in Ente to run on the LAMP stack, and the developers are silent..
